app/src/main/java/com/google/samples/apps/nowinandroid/MainActivity.kt

```diff
@@ -38,16 +58,60 @@ class MainActivity : ComponentActivity() {
     @Inject
     lateinit var lazyStats: dagger.Lazy<JankStats>
 
+    @Inject
+    lateinit var userDataRepository: UserDataRepository
+
     override fun onCreate(savedInstanceState: Bundle?) {
-        installSplashScreen()
+        val splashScreen = installSplashScreen()
         super.onCreate(savedInstanceState)
 
+        /**
+         * The current user data, updated here to drive the UI theme
+         */
+        var userDataResult: Result<UserData> by mutableStateOf(Result.Loading)
+
+        // Update the user data
+        lifecycleScope.launch {
+            lifecycle.repeatOnLifecycle(Lifecycle.State.STARTED) {
+                userDataRepository.userDataStream
+                    .asResult()
+                    .onEach {
+                        userDataResult = it
+                    }
+                    .collect()
+            }
+        }
+
+        // Keep the splash screen on-screen until the user data is loaded
+        splashScreen.setKeepOnScreenCondition {
+            when (userDataResult) {
```

Comment:
Edit: Looked into it and turns out
"The condition is evaluated before each request to draw the application, so it needs to be fast to avoid blocking the UI."
This should be documented here, add comment?

Reply:
Comment added.